front-end/peer-prep/src/components/Auth/SignInPage/AuthButton/SignInButton.tsx

```diff
@@ -0,0 +1,59 @@
+import { IconButton } from "@mui/material";
+
+import * as Styles from "../styles"
+
+import { useNavigate } from "react-router-dom";
+
+import { auth } from "../../Firebase";
+import { GoogleAuthProvider, FacebookAuthProvider, GithubAuthProvider, TwitterAuthProvider, signInWithPopup } from "firebase/auth";
+
+import { useDispatch, useSelector } from "react-redux";
+import * as UserSlice from "../../../redux/reducers/User/UserSlice"
+
+import axios from "axios";
+
+interface SignInButtonProps {
+    provider: GoogleAuthProvider | FacebookAuthProvider | GithubAuthProvider | TwitterAuthProvider;
```

Comment:
Good use of interface to combine all 4 provider into 1 file instead of 4 files with mostly duplicated code.